framework/data/ActiveDataProvider.php

```diff
@@ -159,8 +161,13 @@ protected function prepareTotalCount()
         if (!$this->query instanceof QueryInterface) {
             throw new InvalidConfigException('The "query" property must be an instance of a class that implements the QueryInterface e.g. yii\db\Query or its subclasses.');
         }
-        $query = clone $this->query;
-        return (int) $query->limit(-1)->offset(-1)->orderBy([])->count('*', $this->db);
+        $query = (clone $this->query)->limit(-1)->offset(-1)->orderBy([]);
+        $key = md5((string)$query);
```

Comment:
IMO, this is so ugly. Also, it breaks BC. Look at https://github.com/yiisoft/yii2/issues/20205